sysml.library/Domain Libraries/Analysis/StateSpaceRepresentation.sysml

```diff
@@ -53,23 +57,24 @@ package StateSpaceRepresentation {
                 input = NextState::input,
                 initialState = NextState::stateSpace,
                 timeInterval = NextState::timeStep
-            ) return resultState = result;  
+            ) return resultState = result;
         }
-    }  
+
+        // ContinuousStateSpaceDynamics may cause zero crossings anomaly.
+        event occurrence zeroCrossingEvents[0..*] : ZeroCrossingEventDef;
+    }
 
     /** Computes difference of stateSpace by one timestep, that is x(k+1) - x(k),
          where k is the number of timesteps. */
     abstract calc def Difference(input: Input, stateSpace: StateSpace): StateSpace;
-    // TODO: Computes next stateSpace by adding the diff.  Vector addition should replace this.
-    abstract calc def AddDiff(stateSpace: StateSpace, diff: StateSpace): StateSpace;
 
     /** DiscreteStateSpaceDynamics represents discrete behavior.
       * differences needs to return difference of the stateSpace for each timestep.  */
     abstract action def DiscreteStateSpaceDynamics :> StateSpaceDynamics {
```

Comment:
In order to be more explicit and distinguish from e.g. vector subtraction I would suggest to modify as follows:

- rename `Difference` to `ComputeDifferenceOverTime` or `GetDifferenceOverTime`
- also replace `k is the number of timesteps` by `k is the timestep number`

Reply:
Is OverTime suffix needed?

Reply:
I changed Difference and difference to GetDifference and getDifference, respectively.  Also fixed the comment.